u-boot-update

```diff
@@ -179,19 +190,29 @@ for _KERNEL in ${_KERNELS}; do
 	else
 		_INITRD=""
 	fi
-	if [ -e "${U_BOOT_FDT}" ] && [ -n "${U_BOOT_FDT}" ] && [ "/" = "$(echo "${U_BOOT_FDT}" | head -c1)" ]; then
-		_FDT="fdt ${U_BOOT_FDT}"
-	elif [ -e "${_BOOT_PATH}${U_BOOT_FDT_DIR}${_VERSION}/${U_BOOT_FDT}" ] && [ -n "${U_BOOT_FDT}" ]; then
-		_FDT="fdt ${U_BOOT_FDT_DIR}${_VERSION}/${U_BOOT_FDT}"
-	elif [ -d "${_BOOT_PATH}${U_BOOT_FDT_DIR}${_VERSION}/" ]; then
-		_FDT="fdtdir ${U_BOOT_FDT_DIR}${_VERSION}/"
-	elif [ -e "${_BOOT_PATH}/${U_BOOT_FDT}" ] && [ -n "${U_BOOT_FDT}" ]; then
-		_FDT="fdt /${U_BOOT_FDT}"
-	elif [ -f "${_BOOT_PATH}/${U_BOOT_FDT:-dtb-${_VERSION}}" ] && [ /usr/lib/linux-image- = "${U_BOOT_FDT_DIR}" ]; then
-		_FDT="fdt /${U_BOOT_FDT:-dtb-${_VERSION}}"
-	else
-		_FDT=""
-	fi
+
+	for dtb in ${U_BOOT_FDT}; do
+		# Check if the DTB matches the current kernel version
+		_IMG_VERSION=$(echo "${_KERNEL#*-}" | sed 's/-.*//') 
+		DTB_VERSION=$(echo "$dtb" | sed -n 's/.*-\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/\1/p')
```

Comment:
make this DTB_VERSION=$(echo "$dtb" | sed -n 's/.*-\([0-9]\+\.[0-9]\+\.[0-9]\+-[a-z]\+\).*/\1/p')